libdnf/utils/File.cpp

```diff
@@ -0,0 +1,65 @@
+#include "File.hpp"
+#include "CompressedFile.hpp"
+#include "utils.hpp"
+
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <sstream>
+#include <utility>
+#include <iostream>
+
+extern "C" {
+#   include <solv/solv_xfopen.h>
+};
+
+std::shared_ptr<libdnf::File> libdnf::File::getFile(std::string filePath)
+{
+    if (solv_xfopen_iscompressed(filePath.c_str()) == 1) {
+        return std::make_shared<libdnf::CompressedFile>(filePath);
+    } else {
+        return std::make_shared<libdnf::File>(filePath);
+    }
+}
+
+libdnf::File::File(std::string filePath)
+        : filePath(std::move(filePath))
+{}
+
+void libdnf::File::open(std::ios_base::openmode mode)
+{
+    file.open(filePath, mode);
+}
+
+void libdnf::File::close()
+{
+    file.close();
+}
+
+std::string libdnf::File::read(size_t count)
+{
+    char buffer[count + 1];
+    file.read(buffer, count);
+    return std::string(buffer);
+}
+
+bool libdnf::File::readLine(std::string &line)
+{
+    std::getline(file, line);
+    line += "\n";
+    return !file.eof();
+}
+
+std::string libdnf::File::getContent()
+{
+    std::stringstream ss;
+    std::string line;
+
+    while (readLine(line)) {
```

Comment:
concatenating by line? doesn't look good from perf POV.

Reply:
Optimal solution probably is:
- get file size
- allocate memory
- and then read blocks not lines.ost optimal solution probably is. Get file size, allocate memory and then read blocks not lines.

If you plan reading  from strem without knowing of the its size, you can't allocate memory but use blocks not lines.